File: frontend/src/services/api/order.ts
const API_BASE_URL = "http://localhost:8000";

export interface CheckoutAddressDetail {
  consignee_name: string;
  phone_number: string;
  secondary_phone_number?: string;
  address_type: string;
  address_line_1: string;
  address_line_2?: string;
  city: string;
  state: string;
  pincode: string;
}

export interface CheckoutItemDetail {
  pre_cart_item_id: string;
  combo_name: string;
  size: string;
  qty: number;
  price: number;
  total_price: number;
}

export interface CheckoutSummaryResponse {
  success: boolean;
  items: CheckoutItemDetail[];
  total_items_price: number;
  delivery_charge: number;
  payable_amount: number;
  address: CheckoutAddressDetail;
}

export interface CreateOrderResponse {
  success: boolean;
  order_id: string;
  payable_amount: number;
  razorpay_order_id: string;
  message?: string;
}

export interface OrderItemHistory {
  id: string;
  size: string;
  qty: number;
  price: number;
  combo_name: string;
  description: string;
  accord_1_id: string;
  accord_1_volume_ml: number;
  accord_2_id?: string;
  accord_2_volume_ml?: number;
  accord_3_id?: string;
  accord_3_volume_ml?: number;
}

export interface OrderHistoryItem {
  id: string;
  created_at: string;
  status: string;
  total_amount: number;
  delivery_charge: number;
  payable_amount: number;
  address: CheckoutAddressDetail;
  items: OrderItemHistory[];
}

function getHeaders() {
  const token = localStorage.getItem("jwt_token");
  return {
    "Content-Type": "application/json",
    Authorization: `Bearer ${token}`,
  };
}

export async function fetchCheckoutSummary(
  sessionId: string,
  addressId: string
): Promise<CheckoutSummaryResponse> {
  const res = await fetch(
    `${API_BASE_URL}/orders/checkout-summary?session_id=${sessionId}&address_id=${addressId}`,
    {
      method: "GET",
      headers: getHeaders(),
    }
  );
  if (!res.ok) {
    const errorData = await res.json().catch(() => ({}));
    throw new Error(errorData?.detail || "Failed to fetch checkout summary.");
  }
  return res.json();
}

export async function createOrder(
  sessionId: string,
  addressId: string
): Promise<CreateOrderResponse> {
  const res = await fetch(`${API_BASE_URL}/orders/create`, {
    method: "POST",
    headers: getHeaders(),
    body: JSON.stringify({
      session_id: sessionId,
      address_id: addressId,
    }),
  });
  if (!res.ok) {
    const errorData = await res.json().catch(() => ({}));
    throw new Error(errorData?.detail || "Failed to create order.");
  }
  return res.json();
}

export async function verifyPayment(
  orderId: string,
  razorpayPaymentId?: string,
  razorpaySignature?: string
): Promise<{ success: boolean; message: string }> {
  const res = await fetch(`${API_BASE_URL}/orders/verify-payment`, {
    method: "POST",
    headers: getHeaders(),
    body: JSON.stringify({
      order_id: orderId,
      razorpay_payment_id: razorpayPaymentId,
      razorpay_signature: razorpaySignature,
    }),
  });
  if (!res.ok) {
    const errorData = await res.json().catch(() => ({}));
    throw new Error(errorData?.detail || "Failed to verify payment.");
  }
  return res.json();
}

export async function fetchOrderHistory(): Promise<OrderHistoryItem[]> {
  const res = await fetch(`${API_BASE_URL}/orders/history`, {
    method: "GET",
    headers: getHeaders(),
  });
  if (!res.ok) {
    const errorData = await res.json().catch(() => ({}));
    throw new Error(errorData?.detail || "Failed to fetch order history.");
  }
  return res.json();
}

File: backend/app/schemas/order.py
from typing import Optional, List
from uuid import UUID
from pydantic import BaseModel
from datetime import datetime


# --- Checkout Summary ---
class CheckoutSummaryRequest(BaseModel):
    session_id: UUID
    address_id: UUID


class CheckoutItemDetail(BaseModel):
    pre_cart_item_id: str
    combo_name: str
    size: str
    qty: int
    price: float
    total_price: float


class CheckoutAddressDetail(BaseModel):
    consignee_name: str
    phone_number: str
    secondary_phone_number: Optional[str] = None
    address_type: str
    address_line_1: str
    address_line_2: Optional[str] = None
    city: str
    state: str
    pincode: str


class CheckoutSummaryResponse(BaseModel):
    success: bool
    items: List[CheckoutItemDetail]
    total_items_price: float
    delivery_charge: float
    payable_amount: float
    address: CheckoutAddressDetail


# --- Create Order ---
class CreateOrderRequest(BaseModel):
    session_id: UUID
    address_id: UUID


class CreateOrderResponse(BaseModel):
    success: bool
    order_id: str
    payable_amount: float
    razorpay_order_id: Optional[str] = None
    message: Optional[str] = None


# --- Verify Payment ---
class VerifyPaymentRequest(BaseModel):
    order_id: UUID
    razorpay_payment_id: Optional[str] = None
    razorpay_signature: Optional[str] = None


class VerifyPaymentResponse(BaseModel):
    success: bool
    message: str


# --- Order History ---
class OrderItemHistoryResponse(BaseModel):
    id: UUID
    size: str
    qty: int
    price: float
    combo_name: str
    description: str
    accord_1_id: str
    accord_1_volume_ml: float
    accord_2_id: Optional[str] = None
    accord_2_volume_ml: Optional[float] = None
    accord_3_id: Optional[str] = None
    accord_3_volume_ml: Optional[float] = None


class OrderHistoryResponseItem(BaseModel):
    id: UUID
    created_at: datetime
    status: str
    total_amount: float
    delivery_charge: float
    payable_amount: float
    address: CheckoutAddressDetail
    items: List[OrderItemHistoryResponse]
